public/editor/scripts/project-list.js

```diff
@@ -30,16 +37,57 @@ require(["jquery", "constants", "analytics", "moment"], function($, Constants, a
     return "{{ momentJSLastEdited | safe }}".replace("<% timeElapsed %>", timeElapsed);
   }
 
+  function storageAvailable(type) {
+    try {
+      var storage = window[type],
+      x = '__storage_test__';
+      storage.setItem(x, x);
+      storage.removeItem(x);
+      return true;
+    }
+    catch(e) {
+      return false;
+    }
+  }
+ 
+  var favoriteArray = [];
+
   Array.prototype.forEach.call(projects, function(project) {
     var projectSelector = "#" + project.getAttribute("id");
     var lastEdited = project.getAttribute("data-project-date_updated");
 
+    if(storageAvailable('localStorage')) {
+      var projectId = project.getAttribute("data-project-id");
+      if(favorites.indexOf(projectId) != -1){
+        favoriteArray.push(project);
+        $(projectSelector + " .project-favorite-button").toggleClass("project-unfavorite-button");
```

Comment:
hmm, I think the naming convention here is a bit confusing. How about we use the following toggle:
```html
<!-- Let's assume none of them are favorites by default so a non-favorite button will look something like: -->
<div class="project-favorite-button">Click here to favorite this</div>
```
and
```html
<!-- A favorited project's button will look like: -->
<div class="project-favorite-button project-favorite-selected">Click here to unfavorite this</div>
```
So really, the class you will toggle is `project-favorite-selected`. You can choose to omit the `project-favorite-button` class entirely but I assume you would want that for styling.